Not able to build from master

Hi, 
I just did `git clone https://...` and then `cargo build --release` (with rustup stable) and here's what I have:
```bash
ripp:% cargo build --release                                                                                                                                                                         <master>
   Compiling ark-sipp v0.1.0 (/home/nalos/prog/rust/ripp/sipp)
   Compiling ark-ip-proofs v0.1.0 (/home/nalos/prog/rust/ripp/ip_proofs)
error[E0432]: unresolved import `ark_poly::polynomial::DensePolynomial`
 --> ip_proofs/src/applications/poly_commit/mod.rs:3:5
  |
3 | use ark_poly::polynomial::DensePolynomial as UnivariatePolynomial;
  |     ^^^^^^^^^^^^^^^^^^^^^^---------------^^^^^^^^^^^^^^^^^^^^^^^^
  |     |                     |
  |     |                     help: a similar name exists in the module: `Polynomial`
  |     no `DensePolynomial` in `polynomial`

error[E0432]: unresolved import `ark_poly::polynomial::DensePolynomial`
 --> ip_proofs/src/applications/poly_commit/transparent.rs:3:5
  |
3 | use ark_poly::polynomial::DensePolynomial as UnivariatePolynomial;
  |     ^^^^^^^^^^^^^^^^^^^^^^---------------^^^^^^^^^^^^^^^^^^^^^^^^
  |     |                     |
  |     |                     help: a similar name exists in the module: `Polynomial`
  |     no `DensePolynomial` in `polynomial`

error[E0432]: unresolved import `ark_poly::polynomial::DensePolynomial`
 --> ip_proofs/src/tipa/mod.rs:3:5
  |
3 | use ark_poly::polynomial::DensePolynomial;
  |     ^^^^^^^^^^^^^^^^^^^^^^---------------
  |     |                     |
  |     |                     help: a similar name exists in the module: `Polynomial`
  |     no `DensePolynomial` in `polynomial`

error[E0392]: parameter `F` is never used
  --> ip_proofs/src/applications/poly_commit/transparent.rs:58:32
   |
58 | pub struct BivariatePolynomial<F: Field> {
   |                                ^ unused parameter
   |
   = help: consider removing `F`, referring to it in a field, or using a marker such as `PhantomData`

error[E0392]: parameter `F` is never used
   --> ip_proofs/src/applications/poly_commit/mod.rs:122:32
    |
122 | pub struct BivariatePolynomial<F: Field> {
    |                                ^ unused parameter
    |
    = help: consider removing `F`, referring to it in a field, or using a marker such as `PhantomData`

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0392, E0432.
For more information about an error, try `rustc --explain E0392`.
error: could not compile `ark-ip-proofs`

To learn more, run the command again with --verbose.
```
Here's extract of my `rustup show`:
```bash
active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.48.0 (7eac88abb 2020-11-16)
```

--- a/benches/benches/poly_commit.rs
+++ b/benches/benches/poly_commit.rs
@@ -5,7 +5,9 @@ use ark_ip_proofs::applications::poly_commit::{
     transparent::UnivariatePolynomialCommitment as TransparentIPA,
     UnivariatePolynomialCommitment as IPA, KZG,
 };
-use ark_poly::polynomial::DensePolynomial as Polynomial;
+use ark_poly::polynomial::{
+    univariate::DensePolynomial as UnivariatePolynomial, Polynomial, UVPolynomial,
+};
 
 use csv::Writer;
 use rand::{rngs::StdRng, SeedableRng};
@@ -64,13 +66,9 @@ fn main() {
                 .unwrap();
             csv_writer.flush().unwrap();
             for i in 1..num_trials + 1 {
-                let mut polynomial_coeffs = vec![];
-                for _ in 0..degree + 1 {
-                    polynomial_coeffs.push(<Bls12_381 as PairingEngine>::Fr::rand(&mut rng));
-                }
-                let polynomial = Polynomial::from_coefficients_slice(&polynomial_coeffs);
+                let polynomial = UnivariatePolynomial::rand(degree, &mut rng);
                 let point = <Bls12_381 as PairingEngine>::Fr::rand(&mut rng);
-                let eval = polynomial.evaluate(point.clone());
+                let eval = polynomial.evaluate(&point);
 
                 // Commit
                 start = Instant::now();
@@ -140,13 +138,9 @@ fn main() {
                 .unwrap();
             csv_writer.flush().unwrap();
             for i in 1..num_trials + 1 {
-                let mut polynomial_coeffs = vec![];
-                for _ in 0..degree + 1 {
-                    polynomial_coeffs.push(<Bls12_381 as PairingEngine>::Fr::rand(&mut rng));
-                }
-                let polynomial = Polynomial::from_coefficients_slice(&polynomial_coeffs);
+                let polynomial = UnivariatePolynomial::rand(degree, &mut rng);
                 let point = <Bls12_381 as PairingEngine>::Fr::rand(&mut rng);
-                let eval = polynomial.evaluate(point.clone());
+                let eval = polynomial.evaluate(&point);
 
                 // Commit
                 start = Instant::now();
@@ -218,13 +212,9 @@ fn main() {
                     .unwrap();
                 csv_writer.flush().unwrap();
                 for i in 1..num_trials + 1 {
-                    let mut polynomial_coeffs = vec![];
-                    for _ in 0..degree + 1 {
-                        polynomial_coeffs.push(<Bls12_381 as PairingEngine>::Fr::rand(&mut rng));
-                    }
-                    let polynomial = Polynomial::from_coefficients_slice(&polynomial_coeffs);
+                    let polynomial = UnivariatePolynomial::rand(degree, &mut rng);
                     let point = <Bls12_381 as PairingEngine>::Fr::rand(&mut rng);
-                    let eval = polynomial.evaluate(point.clone());
+                    let eval = polynomial.evaluate(&point);
 
                     // Commit
                     start = Instant::now();

--- a/benches/benches/poly_commit.rs
+++ b/benches/benches/poly_commit.rs
@@ -5,7 +5,9 @@ use ark_ip_proofs::applications::poly_commit::{
     transparent::UnivariatePolynomialCommitment as TransparentIPA,
     UnivariatePolynomialCommitment as IPA, KZG,
 };
-use ark_poly::polynomial::DensePolynomial as Polynomial;
+use ark_poly::polynomial::{
+    univariate::DensePolynomial as UnivariatePolynomial, Polynomial, UVPolynomial,
+};
 
 use csv::Writer;
 use rand::{rngs::StdRng, SeedableRng};
@@ -64,13 +66,9 @@ fn main() {
                 .unwrap();
             csv_writer.flush().unwrap();
             for i in 1..num_trials + 1 {
-                let mut polynomial_coeffs = vec![];
-                for _ in 0..degree + 1 {
-                    polynomial_coeffs.push(<Bls12_381 as PairingEngine>::Fr::rand(&mut rng));
-                }
-                let polynomial = Polynomial::from_coefficients_slice(&polynomial_coeffs);
+                let polynomial = UnivariatePolynomial::rand(degree);
                 let point = <Bls12_381 as PairingEngine>::Fr::rand(&mut rng);
-                let eval = polynomial.evaluate(point.clone());
+                let eval = polynomial.evaluate(&point);
 
                 // Commit
                 start = Instant::now();

--- a/ip_proofs/src/applications/poly_commit/mod.rs
+++ b/ip_proofs/src/applications/poly_commit/mod.rs
@@ -463,7 +463,7 @@ mod tests {
         let eval_proof =
             TestUnivariatePolyCommitment::open(&srs, &polynomial, &y_polynomial_comms, &point)
                 .unwrap();
-        let eval = polynomial.evaluate(point.clone());
+        let eval = polynomial.evaluate(&point);
 
         // Verify proof
         assert!(TestUnivariatePolyCommitment::verify(

--- a/ip_proofs/src/applications/poly_commit/mod.rs
+++ b/ip_proofs/src/applications/poly_commit/mod.rs
@@ -1,6 +1,8 @@
 use ark_ec::{group::Group, msm::VariableBaseMSM, PairingEngine, ProjectiveCurve};
 use ark_ff::{Field, One, PrimeField, UniformRand, Zero};
-use ark_poly::polynomial::DensePolynomial as UnivariatePolynomial;
+use ark_poly::polynomial::{
+    univariate::DensePolynomial as UnivariatePolynomial, Polynomial, UVPolynomial,
+};
 
 use bench_utils::{end_timer, start_timer};
 use std::marker::PhantomData;
@@ -135,7 +137,7 @@ impl<F: Field> BivariatePolynomial<F> {
         point_x_powers
             .iter()
             .zip(&self.y_polynomials)
-            .map(|(x_power, y_polynomial)| x_power.clone() * y_polynomial.evaluate(y.clone()))
+            .map(|(x_power, y_polynomial)| x_power.clone() * y_polynomial.evaluate(&y.clone()))
             .sum()
     }
 }

--- a/ip_proofs/src/applications/poly_commit/mod.rs
+++ b/ip_proofs/src/applications/poly_commit/mod.rs
@@ -1,6 +1,9 @@
 use ark_ec::{group::Group, msm::VariableBaseMSM, PairingEngine, ProjectiveCurve};
 use ark_ff::{Field, One, PrimeField, UniformRand, Zero};
-use ark_poly::polynomial::DensePolynomial as UnivariatePolynomial;
+use ark_poly::polynomial::{
+    univariate::DensePolynomial as UnivariatePolynomial, Polynomial, UVPolynomial,
+};
+
 
 use bench_utils::{end_timer, start_timer};
 use std::marker::PhantomData;
@@ -135,7 +138,7 @@ impl<F: Field> BivariatePolynomial<F> {
         point_x_powers
             .iter()
             .zip(&self.y_polynomials)
-            .map(|(x_power, y_polynomial)| x_power.clone() * y_polynomial.evaluate(y.clone()))
+            .map(|(x_power, y_polynomial)| x_power.clone() * y_polynomial.evaluate(&y))
             .sum()
     }
 }
@@ -461,7 +464,7 @@ mod tests {
         let eval_proof =
             TestUnivariatePolyCommitment::open(&srs, &polynomial, &y_polynomial_comms, &point)
                 .unwrap();
-        let eval = polynomial.evaluate(point.clone());
+        let eval = polynomial.evaluate(&point);
 
         // Verify proof
         assert!(TestUnivariatePolyCommitment::verify(

--- a/ip_proofs/src/applications/poly_commit/transparent.rs
+++ b/ip_proofs/src/applications/poly_commit/transparent.rs
@@ -1,6 +1,8 @@
 use ark_ec::PairingEngine;
 use ark_ff::{Field, Zero};
-use ark_poly::polynomial::DensePolynomial as UnivariatePolynomial;
+use ark_poly::polynomial::{
+    univariate::DensePolynomial as UnivariatePolynomial, Polynomial, UVPolynomial,
+};
 
 use bench_utils::{end_timer, start_timer};
 use std::marker::PhantomData;
@@ -71,7 +73,7 @@ impl<F: Field> BivariatePolynomial<F> {
         point_x_powers
             .iter()
             .zip(&self.y_polynomials)
-            .map(|(x_power, y_polynomial)| x_power.clone() * y_polynomial.evaluate(y.clone()))
+            .map(|(x_power, y_polynomial)| x_power.clone() * y_polynomial.evaluate(&y))
             .sum()
     }
 }
@@ -394,7 +396,7 @@ mod tests {
         let eval_proof =
             TestUnivariatePolyCommitment::open(&ck, &polynomial, &y_polynomial_comms, &point)
                 .unwrap();
-        let eval = polynomial.evaluate(point.clone());
+        let eval = polynomial.evaluate(&point);
 
         // Verify proof
         assert!(

--- a/ip_proofs/src/applications/poly_commit/transparent.rs
+++ b/ip_proofs/src/applications/poly_commit/transparent.rs
@@ -1,6 +1,8 @@
 use ark_ec::PairingEngine;
 use ark_ff::{Field, Zero};
-use ark_poly::polynomial::DensePolynomial as UnivariatePolynomial;
+use ark_poly::{
+    polynomial::univariate::DensePolynomial as UnivariatePolynomial, Polynomial, UVPolynomial,
+};
 
 use bench_utils::{end_timer, start_timer};
 use std::marker::PhantomData;
@@ -71,7 +73,7 @@ impl<F: Field> BivariatePolynomial<F> {
         point_x_powers
             .iter()
             .zip(&self.y_polynomials)
-            .map(|(x_power, y_polynomial)| x_power.clone() * y_polynomial.evaluate(y.clone()))
+            .map(|(x_power, y_polynomial)| x_power.clone() * y_polynomial.evaluate(&y.clone()))
             .sum()
     }
 }

--- a/ip_proofs/src/applications/poly_commit/transparent.rs
+++ b/ip_proofs/src/applications/poly_commit/transparent.rs
@@ -396,7 +396,7 @@ mod tests {
         let eval_proof =
             TestUnivariatePolyCommitment::open(&ck, &polynomial, &y_polynomial_comms, &point)
                 .unwrap();
-        let eval = polynomial.evaluate(point.clone());
+        let eval = polynomial.evaluate(&point);
 
         // Verify proof
         assert!(

--- a/ip_proofs/src/tipa/mod.rs
+++ b/ip_proofs/src/tipa/mod.rs
@@ -1,6 +1,8 @@
 use ark_ec::{group::Group, msm::FixedBaseMSM, PairingEngine, ProjectiveCurve};
 use ark_ff::{to_bytes, Field, One, PrimeField, UniformRand, Zero};
-use ark_poly::polynomial::DensePolynomial;
+use ark_poly::polynomial::{
+    univariate::DensePolynomial, UVPolynomial,
+};
 use bench_utils::{end_timer, start_timer};
 use digest::Digest;
 use itertools::Itertools;

--- a/ip_proofs/src/tipa/mod.rs
+++ b/ip_proofs/src/tipa/mod.rs
@@ -1,6 +1,6 @@
 use ark_ec::{group::Group, msm::FixedBaseMSM, PairingEngine, ProjectiveCurve};
 use ark_ff::{to_bytes, Field, One, PrimeField, UniformRand, Zero};
-use ark_poly::polynomial::DensePolynomial;
+use ark_poly::{polynomial::univariate::DensePolynomial, UVPolynomial};
 use bench_utils::{end_timer, start_timer};
 use digest::Digest;
 use itertools::Itertools;

